ansible/galaxy_ng
# Init django-flags

Issue: [AAP-35418](https://issues.redhat.com/browse/AAP-35418)

Add django-flags dependancy to Hub component.

verifications:
1. add `MY_FLAG` to `FLAGS` in `galaxy_ng/app/settings.py`
```
FLAGS = {
    "MY_FLAG": []
}
```
2. enable, disable flag and verify state
```
> docker compose -f dev/compose/aap.yaml exec manager /bin/bash
bash-4.4# django-admin enable_flag MY_FLAG
Successfully enabled MY_FLAG

bash-4.4# pulpcore-manager shell
Python 3.11.10 (main, Sep 24 2024, 09:33:51) [GCC 8.5.0 20210514 (Red Hat 8.5.0-22)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.29.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from flags.state import flag_state, enable_flag, disable_flag

In [2]: flag_state('MY_FLAG')
Out[2]: True

In [3]: disable_flag('MY_FLAG')

In [4]: flag_state('MY_FLAG')
Out[4]: False
```



(django-flags in EDA: https://github.com/ansible/eda-server/pull/1135)
